device/virtual/x86_64-kvm_x86_64-r0/virtual_chassis/virtual_chassis.py

```diff
@@ -0,0 +1,293 @@
+#!/usr/bin/env python
+
+import argparse
+import docker
+import json
+import os
+import re
+import sys
+import time
+
+# input: topology.json create/delete nsName
+# create/delete vs instances as part of chassis or one of its neighbors
+# after creation, connect each vs to the chassis bridge
+# after neighbor creation, connect the chassis's LC with neighbor
+# assumptions:
+# 1. connect each vs in the chassis network using inband and vlan4094 connected
+#    to the chassis bridge
+# 2. vs connects to the outer namespace through veth pair connecting to the bridge
+# 3. Each neighbor connection the the LC requires a separate bridge
+# 4. Inorder to make the front panel ports (EthernetX) work in the vs
+#    all the 32 front panel ports requires veth pairs (eth1 to eth32) in the vs
+#    eth0 is for the mgmt port
+
+def runCmd(cmd):
+   #print cmd
+   ret = os.popen(cmd)
+   res = ret.read().strip()
+   if res != "":
+      print cmd
+      print res
+   #print res
+   return res
+
+
+class VirtualChassisTopology(object):
+   def __init__(self, topoFile, oper, ns, src=None, image=None):
+      self.instInfo = {}
+      self.topoFile = topoFile
+      self.oper = oper
+      self.ns = ns
+      self.src = src
+      self.image = image
+      self.nss = "sudo ip netns exec " + self.ns
+      self.brReq = {}
+      self.readyIntfsInInst = {}
+      self.chassBr = "br4chs"
+      self.ctns = {}
+
+   def handleRequest(self):
+      with open(self.topoFile, 'r') as f:
+         virtualTopology = json.load(f)["VIRTUAL_TOPOLOGY"]
+         inst = virtualTopology['chassis_instances']
+         if 'neighbor_instances' in virtualTopology:
+            inst += virtualTopology['neighbor_instances']
+         if self.oper == 'verify':
+            self.verifyVirtualChassis(virtualTopology)
+            return
+         if self.oper == 'create':
```

Comment:
Stick to double-quotes throughout for consistency.

Reply:
ack